### PR TITLE
Exos Exam

### PR DESCRIPTION
**Pull request by jpcosal**

### CODE REVIEW

File: Exam/src/exercice1/somme.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+public class somme {
+	public static int SommeCubes(int elts) throws Exception{
+		int res = 0;
+		
+		if(elts == 0) {
+			throw new Exception("Impossible de calculer le cube de 0");
+		}
+		for(int i = 0; i < elts; i++) {
+			res = (int) (res + Math.pow(i, 3));
```
Comment (anicolas98) [inline Exam/src/exercice1/somme.java]: Surcodage avec le cast ?

File: Exam/src/exercice1/sommeTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+
+class sommeTest {
+
+	@Test
+	void testSommeCube() throws Exception {
+		double expected = 3025, elts = 10;
+		assertEquals(expected, exercice1.somme.SommeCubes((int) elts));
```
Comment (anicolas98) [inline Exam/src/exercice1/sommeTest.java]: double test peut être de trop

File: Exam/src/exercice3/Cercle.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package exercice3;
+
+public class Cercle extends Forme{
+	private double rayon;
+
```
Comment (anicolas98) [inline Exam/src/exercice3/Cercle.java]: besoin d'un constructeur

File: Exam/src/exercice3/Carre.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package exercice3;
+
+public class Carre {
+	private double cote;
+
```
Comment (anicolas98) [inline Exam/src/exercice3/Carre.java]: manque d'un constructeur

File: Exam/src/exercice1/somme.java (hunk 1 above)
Comment (jpcosal) [reply Exam/src/exercice1/somme.java]: Not really. `Math.pow` returns a `double`

File: Exam/src/exercice1/somme.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package exercice1;
+
+public class somme {
+	public static int SommeCubes(int elts) throws Exception{
+		int res = 0;
+		
+		if(elts == 0) {
+			throw new Exception("Impossible de calculer le cube de 0");
```
Comment (jpcosal) [inline Exam/src/exercice1/somme.java]: I believe it is possible...

File: Exam/src/exercice1/sommeTest.java (hunk 2 above)
Comment (jpcosal) [reply Exam/src/exercice1/sommeTest.java]: The test case fails... What does this tell you?

File: Exam/src/exercice2/EventItem.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
+package exercice2;
+
+import java.util.Date;
+
+public class EventItem {
+	Date time;
+	String info;
+	String message;
```
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: Was this in the text of your test??

File: Exam/src/exercice2/EventItem.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package exercice2;
+
+import java.util.Date;
+
+public class EventItem {
+	Date time;
+	String info;
```
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: Where does this come from???

File: Exam/src/exercice2/EventItem.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+	String message;
+	
+	public String toString() {
+		String Message = "";
+		
+		int year = time.getYear();
+		int month = time.getMonth();
+		int dat = time.getDay();
+		int hour = time.getHours();
+		int min = time.getMinutes();
```
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: What's the use of all this if you're not even gonna use it in `Message`?

File: Exam/src/exercice3/Forme.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package exercice3;
+
+public abstract class Forme {
```
Comment (jpcosal) [inline Exam/src/exercice3/Forme.java]: You declare an abstract class but... Where is the abstract method??

File: Exam/src/exercice3/Forme.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+	public static double aireForme;
+	public String description;
```
Comment (jpcosal) [inline Exam/src/exercice3/Forme.java]: I think I've seen this before...

File: Exam/src/exercice3/Forme.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+package exercice3;
+
+public abstract class Forme {
+	public static double aireForme;
```
Comment (jpcosal) [inline Exam/src/exercice3/Forme.java]: Seriously...

File: Exam/src/exercice3/Forme.java [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+	public String description;
+	
+	public static double calculAireForme() {
```
Comment (jpcosal) [inline Exam/src/exercice3/Forme.java]: Once more, seriously?

File: Exam/src/exercice3/Cercle.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
+package exercice3;
+
+public class Cercle extends Forme{
+	private double rayon;
+	
+	public double Calcaire() {
+		double aireCercle = 0;
+		
+		aireCercle = (Math.PI*rayon*2);
```
Comment (jpcosal) [inline Exam/src/exercice3/Cercle.java]: :fearful:

**Pull request closed without merging**